shared/reducers/wallets.js

```diff
@@ -26,19 +26,17 @@ export default function(state: Types.State = initialState, action: WalletsGen.Ac
     case WalletsGen.clearBuiltPayment:
       return state.set('builtPayment', Constants.makeBuiltPayment())
     case WalletsGen.paymentDetailReceived:
-      return state.updateIn(['paymentsMap', action.payload.accountID], payments =>
-        payments.update(payments.findIndex(p => p.id === action.payload.paymentID), payment =>
-          payment.merge({
-            publicMemo: action.payload.publicMemo,
-            publicMemoType: action.payload.publicMemoType,
-            txID: action.payload.txID,
-          })
-        )
+      return state.updateIn(['paymentsMap', action.payload.accountID], (payments = I.Map()) =>
+        Constants.updatePaymentMap(payments, [action.payload.payment])
       )
     case WalletsGen.paymentsReceived:
       return state
-        .setIn(['paymentsMap', action.payload.accountID], I.List(action.payload.payments))
-        .setIn(['pendingMap', action.payload.accountID], I.List(action.payload.pending))
+        .updateIn(['paymentsMap', action.payload.accountID], (paymentsMap = I.Map()) =>
```

Comment:
prefer
```
.updateIn(['paymentsMap', action.payload.accountID], I.Map(), paymentsMap =>
```

Reply:
I got turned off of `notSetValue` making this PR because of [this](https://facebook.github.io/immutable-js/docs/#/Map/updateIn):

> If the `updater` function returns the same value it was called with, then no change will occur. This is still true if `notSetValue` is provided.
> For code using ES2015 or later, using `notSetValue` is discourged in favor of function parameter default values.

We use `null` to mean not loaded and an empty `Map` to mean loaded and empty, so that was causing a problem